signatures/helpers/arguments_helpers.go

```diff
@@ -17,9 +17,23 @@ func GetTraceeArgumentByName(event trace.Event, argName string) (trace.Argument,
 	return trace.Argument{}, fmt.Errorf("argument %s not found", argName)
 }
 
+// GetInitializedTraceeArgumentByName is an envelope function for GetTraceeArgumentByName that
+// return error if the argument is not initialized.
+func GetInitializedTraceeArgumentByName(event trace.Event, argName string) (trace.Argument, error) {
```

Comment:
I think it might be better to actually use a "query options struct" in the previous function to change its behavior rather than exporting a new function (internally we can keep these as two `private` separate functions.

So we would have one function `GetTraceeArgument(event, name, opts)` where opts is a struct that could for now look like `{ DefaultValue bool //document this option` (we could perhaps add other query options later), and it would internally call either this function (which we would make private) or the old function.

Reply:
So if we follow the Clean Code rules, then:
`Don't use flag arguments. Split method into several independent methods that can be called from the client without the flag.`

I tried to follow this rule, but if you think it is more useable this way I have no problem with changing it.

Reply:
> So if we follow the Clean Code rules, then:
Don't use flag arguments. Split method into several independent methods that can be called from the client without the flag.

I would usually also follow this rule, however we now have to take into account that we have to make sure that non tracee devs can, will and should use these helpers. As such we have to make their correct use as clear as possible, and an options struct would encode it well IMO.
With the current naming schema I think if I were not a tracee dev it wouldn't be clear to me what's the difference between the two and why there is a difference as well (this is also a documentation issue).